### PR TITLE
[asm] add tests for session fingerprints

### PR DESCRIPTION
**Pull request by manuel-alvarez-alvarez**
## Motivation

Include new session fingeprints in the tests.

## Changes

Adds a new test to validate the generation of session fingerprints, it also updates the java springboot weblog with the new endpoint.

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

**Review by robertomonteromiguel — APPROVED** [review]

**Comment by simon-id** [comment]
It doesn't validate the RC capability that goes with it no ?

**Review by smola — APPROVED** [review]

File: tests/appsec/test_fingerprinting.py [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import requests
```
Comment (cbeauchesne) [inline tests/appsec/test_fingerprinting.py:1]: typo ?

File: utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/App.java [hunk 2]
```diff
@@ -158,6 +162,19 @@ String postWafXml(@RequestBody XmlObject object) {
         return object.toString();
     }
 
+    @GetMapping(value = "/session/new")
```
Comment (cbeauchesne) [inline utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/App.java:165]: Could you document these new endpoints in `docs/weblog`? 🙏

**Review by cbeauchesne — CHANGES_REQUESTED** [review]

**Comment by manuel-alvarez-alvarez** [comment]
> It doesn't validate the RC capability that goes with it no ?

No, not yet, none of the fingerprint RCs are validated, but my plan is to do another PR as soon as the RASP capabilities PR is merged.

File: tests/appsec/test_fingerprinting.py (hunk 1 above)
Comment (manuel-alvarez-alvarez) [reply tests/appsec/test_fingerprinting.py]: Yep, removed, thanks!

File: utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/App.java (hunk 2 above)
Comment (manuel-alvarez-alvarez) [reply utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/App.java:171]: Yep, completely forgot that we have that documentation, updated, thanks!

**Comment by cbeauchesne** [comment]
I've updated your branch, to fix some CI failures

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**